synapse/handlers/federation.py

```diff
@@ -1854,10 +1863,13 @@ def _persist_auth_tree(self, origin, auth_events, state, event):
             # invalid, and it would fail auth checks anyway.
             raise SynapseError(400, "No create event in state")
 
-        room_version = create_event.content.get(
+        room_version_id = create_event.content.get(
             "room_version", RoomVersions.V1.identifier
         )
 
+        if room_version.identifier != room_version_id:
+            raise SynapseError(400, "Room version mismatch")
```

Comment:
I'm not sure that 400 is the right response code when the remote server lies to us, but :man_shrugging:

Reply:
It's the closest error code to "please bugger off and never sully our doors with that request again" that I know of :)